### PR TITLE
more modifications

### PR DESCRIPTION
**Pull request by LinguList**
Here, again, you see how we can expose the new FormSpec in the chen data, in order to be more transparent. Apparently, we have all default sin this example, but it is useful to keep in mind, and to learn how to actually trigger different behavior.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,13 +29,13 @@ This dataset comprises 25 Hmong-Mien varieties, which were originally digitized
 
 - **Varieties:** 25
 - **Concepts:** 883
-- **Lexemes:** 21,573
+- **Lexemes:** 21,967
```
Comment (LinguList) [inline README.md]: Note that I switched to a splitting dataset, which explains the increase in words.

File: lexibank_chenhmongmien.py [hunk 2]
```diff
@@ -27,12 +27,19 @@ class HLanguage(Language):
...
 
 
-class Dataset(NonSplittingDataset):
+class Dataset(MyDataset):
     dir = Path(__file__).parent
     id = "chenhmongmien"
     concept_class = HConcept
     language_class = HLanguage
 
+    form_spec = FormSpec(
```
Comment (LinguList) [inline lexibank_chenhmongmien.py]: Important for all dataset that tinker with forms (i.e., most of them): do not overwrite, do not split yourself, just use the pre-defined parameters here. And use with care, in any case.

**Comment by chrzyki** [comment]
Thanks, can merge and resolve the conflicts if that's okay with you?

File: lexibank_chenhmongmien.py [hunk 3]
```diff
@@ -77,18 +76,17 @@ def cmd_makecldf(self, args):
 
         args.writer.add_languages()
 
-        languages = collections.OrderedDict([(k['Name'], k['ID']) for k in self.languages])
+        languages = OrderedDict([(k['Name'], k['ID']) for k in self.languages])
         args.writer.add_sources(*self.raw_dir.read_bib())
         missing = {}
-        for cgloss, entry in tqdm(enumerate(data), desc='cldfify the data', total=len(data)):
+        for cgloss, entry in pb(enumerate(data), desc='cldfify the data', total=len(data)):
```
Comment (tresoldi) [inline lexibank_chenhmongmien.py]: pb -> progressbar?

**Comment by chrzyki** [comment]
Merged manually in https://github.com/lexibank/chenhmongmien/commit/7565c3daf9de52e1662a898bda26e048ae7e787f

**Comment by chrzyki** [comment]
Sorry @tresoldi saw your comments too late.

File: lexibank_chenhmongmien.py [hunk 4]
```diff
@@ -27,12 +27,19 @@ class HLanguage(Language):
...
+    form_spec = FormSpec(
+            missing_data=['*', '---', '-'],
+            separators=";/,",
+            strip_inside_brackets=True,
+            brackets={"(": ")"}
+            )
+
     def cmd_download(self, args):
         with self.raw_dir.temp_download(
             "https://en.wiktionary.org/wiki/Appendix:Hmong-Mien_comparative_vocabulary_list",
```
[the remark this answers is not in the document]
Comment (LinguList) [reply lexibank_chenhmongmien.py]: Please make PR later. I didn't even test that part of the code, and I am not willing to do so, as I have been spending the whole day on working on language names, with this PR in fact being here for over a week now with nobody noticing.

**Comment by tresoldi** [comment]
I am not sure what happened here, I had the original PR ready 24 days ago (you can tell by the way I am referring to FormSpecs as something new), the only comment I made today was about the progressbar.

**Comment by LinguList** [comment]
I guess you did not press on SEND off of the review. never mind.

**Pull request closed without merging**